### DISCUSSION TITLE
How to use rsbuild with outputFileSystem?

### DISCUSSION THREAD

**Discussion by dartess** [web-infra-dev/rsbuild#2530] [General]
What I have: `RsbuildConfig` and the desire to get the build result in a variable, without writing files to disk.

`rspack` has the ability to specify a custom file system for the output with `outputFileSystem` option: https://www.rspack.dev/api/javascript-api#custom-file-systems.

I tried something like this, but when I add the `compiler` arg I get a build error `Rspack build failed!` (`Resolve error: Can't resolve './src' in ...`).

```js
  const rsbuildInstance = await createRsbuild({ rsbuildConfig });
  const fs = createFsFromVolume(new Volume());
  const compiler = rspack({});
  compiler.outputFileSystem = fs;
  await rsbuildInstance.build({ compiler });
```

**Comment by chenjiahan** [comment]
Rsbuild provide a `onAfterCreateCompiler` API to access the compiler object, see: https://rsbuild.dev/api/javascript-api/instance#rsbuildonaftercreatecompiler

```js
import { createRsbuild } from '@rsbuild/core';

const rsbuild = await createRsbuild({ rsbuildConfig });
const fs = createFsFromVolume(new Volume());

rsbuild.onAfterCreateCompiler(({ compiler }) => {
  compiler.outputFileSystem = fs;
});
```
